### PR TITLE
[SkeletonPage] Rebuild with layout primitives

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes [#7714](https://github.com/Shopify/polaris/issues/7714)

### WHAT is this pull request doing?

- Refactors `SkeletonPage` and its child components to use our layout primitives.
- Removes `max-width: 100%` property from AlphaStack
- Adds storybook examples for `SkeletonPage` with `narrowWidth` and with `fullWidth` props

#### Issues with current PR:

   <details>
      <summary>Unable to center `SkeletonPage` when `narrowWidth` prop is used - SOLVED ✅</summary>
<table>
<tr>
<td> Before </td> <td> After </td>
</tr>
<tr>
<td> Screenshot 2022-11-24 at 9 13 21 AM </td>
<td> Screenshot 2022-11-24 at 9 10 25 AM </td>
</tr>
<tr>
<td> Screenshot 2022-11-24 at 9 13 42 AM </td>
<td> Screenshot 2022-11-24 at 9 11 17 AM </td>
</tr>
</table>
    </details>

<details>
      <summary>Our current `Text` component has the opposite breakpoint behaviour compared to how `.Title` is used in `SkeletonPage` - SOLVED ✅</summary>
      <table>
<tr>
<td> Before </td> <td> After </td>
</tr>
<tr>
<td> before text </td>
<td> before text </td>
</tr>
<tr>
<td> 

```java
.Title {
  font-weight: var(--p-font-weight-semibold);
  font-size: 24px;
  line-height: 28px;

  @media #{$p-breakpoints-md-up} {
    font-size: 20px;
  }
}
```

 </td>
<td>
    
```java
.headingXl {
  font-size: var(--p-font-size-300);
  line-height: var(--p-font-line-height-3);

  @media #{$p-breakpoints-md-up} {
    font-size: var(--p-font-size-400);
    line-height: var(--p-font-line-height-4);
  }
}
```
    
</td>
</tr>
</table>
    </details>

~~`role` prop is missing from `Box`, creating a separate issue to solve this [here](https://github.com/Shopify/polaris/pull/7799)~~ SOLVED ✅

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by aveline** [comment]
@chaz I created a codepen of how we should be able to use flex to center when `narrowWidth` is set

https://codepen.io/enileva/pen/xxzjOej

then I tried it with the component by wrapping the outer `Box` with this `div`

`<div style={{display: 'flex', flexDirection: 'column', alignItems: 'center'}}>`

which worked

localhost_6006__path=_story_all-components-skeletonpage--with-narrow-width

but as we know, using `AlphaStack` doesn't. And then I realized it's because of the child `AlphaStack` with the `fullWidth`, the styles are getting mixed up with the parent `AlphaStack` that we don't want to be `fullWidth`. So that's a problematic implementation with `fullWidth` that we should revisit. Any thoughts @kyledurand?

Let's try to figure out if we can fix `AlphaStack` but if that drags on, we can unblock this by shipping it with the div and inline styles above?

**Comment by aveline** [comment]
I looked at `AlphaStack` and this is the issue

https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/AlphaStack/AlphaStack.scss#L30

It's overriding the `max-width` on `Box`. So not related to `fullWidth` like I originally thought.

@laurkim @kyledurand any context on why we added `max-width: 100%;` to `AlphaStack`? Would it be safe to remove?

**Comment by kyledurand** [comment]
> any context on why we added max-width: 100%; to AlphaStack? Would it be safe to remove?

Might have something to do with the `> * {width: 100%}` for fullWidth below. I think maybe it was causing overflow on the container unless max width was applied but I'm speculating. Lo would know when she's back.

For now, as an unblocker, we could remove the max width and do:

```scss
.fullWidth {
  > * {
    flex: 1 1 auto;
  }
}
```

And see if that causes any visual regressions. We shouldn't need to use width on a flex container these days.

**OR**

Try removing the fullWidth prop because I'm not sure it's used anywhere and it's kind of misleading. The flex container should take the full width anyway and messing with the ~~children~~ direct descendants in this way feels weird 
image

**Comment by laurkim** [comment]
> It's overriding the `max-width` on `Box`. So not related to `fullWidth` like I originally thought.
> 
> @laurkim @kyledurand any context on why we added `max-width: 100%;` to `AlphaStack`? Would it be safe to remove?

@aveline @kyledurand I think it was added to fix a bug on content when the `fullWidth` prop was passed in during the early stages of build. Should be safe to remove.

@chazdean can we add separate storybook examples for SkeletonPage with `narrowWidth` and with `fullWidth` props?

File: .changeset/strong-bottles-swim.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/polaris': patch
+---
+
+Refactored `SkeletonPage` to use primitive Layout components
+Removed `max-width: 100%` from AlphaStack
+Added `narrowWidth` and `fullWidth` examples to AlphaStack stories
```
Comment (laurkim) [inline .changeset/strong-bottles-swim.md]:
```suggestion
'@shopify/polaris': minor
---

Refactored `SkeletonPage` to use primitive Layout components
Removed `max-width` on children in `AlphaStack`
Added `narrowWidth` and `fullWidth` examples to `AlphaStack` stories
```

**Comment by sarahill** [comment]
> Our current `Text` component has the opposite breakpoint behaviour compared to how `.Title` is used in `SkeletonPage`

Yea, this is one I expected we'd have to manually set as it does behave differently. Expected behavior: Page title remains `20px` across breakpoints.

image

**Review by sarahill — APPROVED** [review]
✅

File: polaris-react/src/components/SkeletonPage/SkeletonPage.tsx [hunk 2]
```diff
@@ -31,46 +33,71 @@ export function SkeletonPage({
...
+    <AlphaStack align="center" fullWidth>
+      <Box
+        padding="0"
+        paddingInlineStart={{sm: '6'}}
+        paddingInlineEnd={{sm: '6'}}
+        maxWidth="998px"
+        aria-label={i18n.translate('Polaris.SkeletonPage.loadingLabel')}
+        role="status"
+        {...(narrowWidth && {
+          maxWidth: '662px',
```
Comment (kyledurand) [inline polaris-react/src/components/SkeletonPage/SkeletonPage.tsx]: I think these should be moved to the scss file. If we do something like:

```scss
--pc-skeleton-page-max-width: 998px;
--pc-skeleton-page-max-width-narrow: 662px;
```
Then the px values get converted to rem with post css.

Then you can just pass in `var(--pc-skeleton-page-max-width), var(--pc-skeleton-page-max-width-narrow)` as the props

**Review by laurkim — DISMISSED** [review]
I recommend decoupling the new full width and narrow width stories from this PR and just merging those into main directly, then rebasing this and `layout-rebuild-batch-2` to pick up the changes (or dropping the commit to add those stories).
I tested the existing SkeletonPage without any of the layout primitive changes off latest main with those new stories added and I'm seeing a difference for the `withFullWidth` example. Here's a gif comparing the chromatic from this PR and me running latest main with the story locally.

skeletonpage

Also, looks like this is exceeding size limit. Can we bump to the threshold to 220 kb?

**Review by aveline — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
Tophatted stories at all breakpoints against latest main running locally and lgtm! Nice 💯

**Pull request merged**